### ISSUE TITLE
How to load external station data and GCM data

### ISSUE THREAD

**Issue by EboQuabena** [metno/esd#267]
I have been trying to use the 'esd' r-package to downscale CMIP5 data to station data in Ghana at latitude = 5.783 and longitude = 0.633 to make a projection. I have a little difficulty loading my external station data and GCM data unto the package. 

Can you be of help to me in this regard?

**Comment by brasmus** [comment]
You need to download the CMIP data onto your hard drive - see our wiki-page on this GitHub site for tips. I suggest you download data for a region that covers 0-10N and 5W-5E so that you get a sufficient representation of the large-scale conditions.

**Comment by EboQuabena** [comment]
Is it possible to get a simple script to help me downscale a gcm projection data to that of one station data? I have been trying to do that but to no avail

**Comment by brasmus** [comment]
Yes, but it's **not advised** to do it that way. It's far better to **downscale a group of stations** within a small region for **a large ensemble of GCMs**. One GCM will likely give you misleading results and could result in maladaptation - it's a big risk to do it that way.

**Comment by EboQuabena** [comment]
Exactly what I want to do. I’ll be using an ensemble of GCMs (mainly cmip6 projection dataset for ssp245, ssp370 and ssp585.

I wanted to use one GCM and one station to try and get the downscaling working. I’ll then proceed with the ensemble. 

Also, the outputs of map(X) , plot (x) and plot (y) give me sensible results.

**Comment by brasmus** [comment]
Very good! So if your station data is the data object `y` and your GCM data `X`, then you need to compute the EOFs first `x <- EOF(X)`. Then the simple downscaling exercise is `z <- DS(y,x,verbose=TRUE)` using the EOFs as predictor. Here I set the argument `verbose=TRUE` to get more information about the process and an idea about how far the code reaches before there is a problem. A typical problem is if the station and the GCM data have different time scales, e.g. daily and monthly. One way to ensure similar timescales is e.g. using  `z <- DS(as.seasonal(y),as.seasonal(EOF(X)),verbose=TRUE)`. Sometimes the objects use different format for time, e.g. date or just year. Then you can fix this by `index(y) <- year(y)` if you have e.g. annual data or aggregated data for one season. One way to check is to write `class(y)` and `range(index(y))`. It's a bit complicated because there are so many different ways to provide data, so sometimes you need to add a couple of extra lines...

**Comment by EboQuabena** [comment]
Great! The challenge now is to get the future downscaled data. The results of the downscaling of the GCM end where the station data ends. How do I get it to continue to 2100?

**Comment by brasmus** [comment]
I guess you downscaled using the reanalysis for calibration. To downscale a GCM, then you need to use 'common EOFs' where you combine the reanalysis and the GCM data into a joint data object and compute an EOF for this combined dataset (this is called 'common EOFs'). Then you repeat the downscaling with this common EOF rather than the ordinary EOF representing the reanalysis only.
